### PR TITLE
Add Ballot Printing

### LINKED ISSUE #63: 🖨️ Export Ballot to PDF
**Type**: client
**FOF**:  2

## Description

- Create a template for the PDF download for the ballot pages and export the data of the user's selections into it
- Use https://react-pdf.org/

## Screenshot

image

## Todo

- [ ] Setup PDF template for ballot
- [ ] Export ballot info into PDF template

## Links
- https://react-pdf.org/

### PR DESCRIPTION
**Pull request by code-arman**
## Status:

:rocket: Ready

## Summary

Allow user to download PDF of selected candidates on fav candidates page

Closes #63, #109

## Changes

- Add borders to all buttons
- Fix ballot printing
## Screenshots
Check out the deployment 💯

### CODE REVIEW

File: client/src/components/ui/ElectionLabel.tsx [hunk 1]
```diff
@@ -12,15 +12,19 @@ interface ElectionLabelProps extends TextProps {
...
 
   const elecId =
     electionId ??
     Number(window.location.pathname.match(/\/ballot\/(\d+)\/.*/)?.at(1));
 
   const election = elections.find((e) => Number(e.id) === Number(elecId));
 
+  if (isLoading) {
+    return null;
+  }
```
Comment (amit-sawhney) [inline client/src/components/ui/ElectionLabel.tsx:26]: Hmmm maybe a fragment over null?

**Review by amit-sawhney — APPROVED** [review]
The functionality works. we will clean this later

File: client/src/api/ballotReady.ts [hunk 2]
```diff
@@ -27,7 +27,7 @@ const instance = axios.create({
   },
 });
 
-const USING_FAKE_DATA = false;
+const USING_FAKE_DATA = true;
```
Comment (ayanmallik) [inline client/src/api/ballotReady.ts:30]: Change this to false

File: client/src/components/pdf/BallotPrint.tsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            ) : (
+              <></>
+            )}
+
+            {Object.entries(election).map(([level, raceMap]) => {
+              const hasCandidates =
+                Object.values(raceMap).reduce(
+                  (a, b) => a + b.candidates.length,
+                  0,
+                ) > 0;
```
Comment (ayanmallik) [inline client/src/components/pdf/BallotPrint.tsx:107]: Nice job using reduce here, Arman

File: client/src/data/positions.json [hunk 4]
```diff
@@ -454,7 +454,7 @@
       "candidates": [
         {
           "first_name": "Cameron",
-          "last_name": "Botsford",
+          "last_name": "Botkvjzxcvklzxcjvklxcjvlkxcjvkvjski",
```
Comment (ayanmallik) [inline client/src/data/positions.json:457]: Way to test out those edge cases, Arman

File: client/src/pages/tab/BallotContainer.tsx [hunk 5]
```diff
@@ -26,16 +28,20 @@ export const BallotContainer: React.FC<BallotContainerProps> = ({
...
       {...rest}
     >
-      {children}
+      {isLoading ? (
+        <Center h="60vh">
+          <Spinner size="xl" />
+        </Center>
+      ) : (
+        children
+      )}
```
Comment (ayanmallik) [inline client/src/pages/tab/BallotContainer.tsx:43]: Such a great idea using the Ballot Container to house all of the loading state in the ballot

File: client/src/theme/foundations/colors.ts [hunk 6]
```diff
@@ -195,7 +195,7 @@ export const BrinkColors: Colors = {
     800: '#44080d',
     900: '#1e0001',
   },
-  greenParty: '#317500',
+  greenParty: '#2D6B00',
```
Comment (ayanmallik) [inline client/src/theme/foundations/colors.ts:198]: Nice job using high contrast colors!

**Review by ayanmallik — APPROVED** [review]
Amazing work, Arman. World record for one of the slowest PR's, but the functionality is AWESOME! Polling place workers are going to fall in love with this!
